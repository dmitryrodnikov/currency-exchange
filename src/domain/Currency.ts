export enum Currency {
    USD = 'USD',
    EUR = 'EUR',
    GBP = 'GBP',
}

export const Currencies = Object.keys(Currency);
