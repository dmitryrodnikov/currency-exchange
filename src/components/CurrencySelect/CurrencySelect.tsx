import * as React from 'react';
import styled from 'styled-components';
import {Currency} from '../../domain/Currency';
import {Button, ButtonClickHandler} from '../Button/Button';

const Wrapper = styled.div`
    display: flex;
    margin: 0 -2px;
`;

interface CurrencySelectorProps {
    currencies: Currency[];
    selected: Currency;
    onItemClick: ButtonClickHandler<Currency>;
}

export function CurrencySelect(props: CurrencySelectorProps) {
    return (
        <Wrapper>
            {
                props.currencies.map(currency => {
                    return (
                        <Button
                            value={currency}
                            key={currency}
                            active={currency === props.selected}
                            onClick={props.onItemClick}
                        >
                            {currency}
                        </Button>
                    );
                })
            }
        </Wrapper>
    );
}
