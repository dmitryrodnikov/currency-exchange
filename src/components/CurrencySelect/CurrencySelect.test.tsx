import React from 'react';
import {mount, render, shallow} from 'enzyme';
import {CurrencySelect} from './CurrencySelect';
import {Currency} from '../../domain/Currency';
import {spy} from 'sinon';

const CURRENCIES= [Currency.USD, Currency.GBP, Currency.EUR];

describe('CurrencySelect', () => {
    it('Shallow render snapshots are equal', () => {
        const wrapper = shallow(
            <CurrencySelect
                currencies={CURRENCIES}
                selected={Currency.USD}
                onItemClick={() => {}}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('Render snapshots are equal', () => {
        const wrapper = render(
            <CurrencySelect
                currencies={CURRENCIES}
                selected={Currency.USD}
                onItemClick={() => {}}
            />
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('Click handler called once with exact argument', () => {
        const clickHandler = spy();
        const wrapper = mount(
            <CurrencySelect
                currencies={CURRENCIES}
                selected={Currency.GBP}
                onItemClick={clickHandler}
            />
        );

        wrapper.find('div').at(1).simulate('click');

        expect(clickHandler.calledOnceWithExactly(Currency.USD)).toBe(true);
    });
});
