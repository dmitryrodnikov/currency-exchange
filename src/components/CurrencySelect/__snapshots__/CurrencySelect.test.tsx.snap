// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrencySelect Render snapshots are equal 1`] = `
<div
  class="sc-bwzfXH cVDcWW"
>
  <div
    class="sc-bdVaJa gMmIAm"
  >
    USD
  </div>
  <div
    class="sc-bdVaJa fREQqs"
  >
    GBP
  </div>
  <div
    class="sc-bdVaJa fREQqs"
  >
    EUR
  </div>
</div>
`;

exports[`CurrencySelect Shallow render snapshots are equal 1`] = `
<styled.div>
  <Button
    active={true}
    key="USD"
    onClick={[Function]}
    value="USD"
  >
    USD
  </Button>
  <Button
    active={false}
    key="GBP"
    onClick={[Function]}
    value="GBP"
  >
    GBP
  </Button>
  <Button
    active={false}
    key="EUR"
    onClick={[Function]}
    value="EUR"
  >
    EUR
  </Button>
</styled.div>
`;
